crdt_test.go

```diff
@@ -0,0 +1,443 @@
+package crdt
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	cid "github.com/ipfs/go-cid"
+	ds "github.com/ipfs/go-datastore"
+	query "github.com/ipfs/go-datastore/query"
+	dssync "github.com/ipfs/go-datastore/sync"
+	dstest "github.com/ipfs/go-datastore/test"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	ipld "github.com/ipfs/go-ipld-format"
+	log "github.com/ipfs/go-log"
+	"github.com/ipfs/go-merkledag"
+	mdutils "github.com/ipfs/go-merkledag/test"
+)
+
+var numReplicas = 50
+var debug = false
+
+func init() {
+	_ = log.SetDebugLogging
+	log.SetLogLevel("crdt", "debug")
+	rand.Seed(time.Now().UnixNano())
+}
+
+type testLogger struct {
+	name string
+	l    log.StandardLogger
+}
+
+func (tl *testLogger) Debug(args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Debug(args...)
+}
+func (tl *testLogger) Debugf(format string, args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Debugf("%s "+format, args...)
+}
+func (tl *testLogger) Error(args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Error(args...)
+}
+func (tl *testLogger) Errorf(format string, args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Errorf("%s "+format, args...)
+}
+func (tl *testLogger) Fatal(args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Fatal(args...)
+}
+func (tl *testLogger) Fatalf(format string, args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Fatalf("%s "+format, args...)
+}
+func (tl *testLogger) Info(args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Info(args...)
+}
+func (tl *testLogger) Infof(format string, args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Infof("%s "+format, args...)
+}
+func (tl *testLogger) Panic(args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Panic(args...)
+}
+func (tl *testLogger) Panicf(format string, args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Panicf("%s "+format, args...)
+}
+func (tl *testLogger) Warning(args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Warning(args...)
+}
+func (tl *testLogger) Warningf(format string, args ...interface{}) {
+	args = append([]interface{}{tl.name}, args...)
+	tl.l.Warningf("%s "+format, args...)
+}
```

Comment:
testLogger seems like something that can move to a separate file, say `test_util.go`

Reply:
I would not move it to a non test file. Moving it to a separate test file makes me connsider moving also mockBroadcaster or mockDAGSync. In the end I don't think there is any significant advantage in having it live in a separate place (and it makes you need to open more files).